src/backend/distributed/planner/multi_planner.c

```diff
@@ -837,6 +874,27 @@ CurrentJoinRestrictionContext(void)
 
 
 /*
+ * CurrentRestrictionsMemoryContext returns the the last restriction's memory context
+ * from the list.
+ */
+static MemoryContext
+CurrentRestrictionsMemoryContext(void)
+{
+	PlannerRestrictionContext *plannerRestrictionContext = NULL;
+	MemoryContext restrictionsMemoryContext = NULL;
+
+	Assert(plannerRestrictionContextList != NIL);
+
+	plannerRestrictionContext =
+		(PlannerRestrictionContext *) linitial(plannerRestrictionContextList);
+
+	restrictionsMemoryContext = plannerRestrictionContext->restrictionsMemoryContext;
+
+	return restrictionsMemoryContext;
+}
```

Comment:
Why is this function needed?   The callers already use CurrentJoinRestrictionContext()/CurrentRelationRestrictionContext().

I think it'd make more sense to replace CurrentJoinRestrictionContext()/CurrentRelationRestrictionContext() with CurrentRestrictionContext(), and then have the individual functions access ->relationRestrictionContext/joinRestrictionContext/restrictionsMemoryContext;

@onderkalaci Why did you add those separate functions?